### PR TITLE
Module/rainstorm/1.0 short

### PR DESCRIPTION
**Pull request by Kdreval**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [X] I used the cookiecutter template and updated the placeholder rules.

- [X] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules). 

  - [X] All references to the `rules` object (_e.g._ for input files) are wrapped with `str()`.

- [X] Every rule in the module is either listed under `localrules` or has the `threads` and `resources` directives.

- [X] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [X] I updated the final target rule (`*_all`) to include every output rule.

- [X] I explained important module design decisions in `CHANGELOG.md`.

- [X] I tested the module on real data for all supported `seq_type` values.

- [X] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [X] I did not set any global wildcard constraints. Any/all wildcard constraints are set on a per-rule basis. 

- [X] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository).

- [X] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [X] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### If applicable

- [X] I added more granular output subdirectories.

- [ ] I added rules to the `reference_files` workflow to generate any new reference files.

- [ ] I added subdirectories with large intermediate files to the list of `scratch_subdirectories` in the `default.yaml` configuration file.

- [X] I updated the list of available wildcards for the input files in the `default.yaml` configuration file.

## Checklist for Updated Module

To be completed.

### CODE REVIEW

**Comment by lkhilton** [comment]
@Kdreval I have reviewed this and have a couple of suggestions for your consideration!

**Comment by lkhilton** [comment]
One more thought: Since this module doesn't explicitly use the samples table, maybe it would be a good idea to drop the base schema from this module. If you decide to implement a table of genome_build-cohort pairs you could write a schema that has those as required fields.

File: modules/rainstorm/1.0/rainstorm.smk [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        rainstorm = str(rules._rainstorm_install.output.rainstorm),    
+        maf = str(rules._rainstorm_input_maf.output.maf),
+        index_subset = str(rules._rainstorm_subset_index.output.index_subset)
+    output:
+        complete = CFG["dirs"]["rainstorm"] + "{genome_build}/{cohort_name}_out_background_100k_binned_density.tsv"
+    log:
+        stdout = CFG["logs"]["rainstorm"] + "{genome_build}/{cohort_name}_rainstorm.stdout.log",
+        stderr = CFG["logs"]["rainstorm"] + "{genome_build}/{cohort_name}_rainstorm.stderr.log"
+    params:
+        out_name = CFG["dirs"]["rainstorm"] + "{genome_build}/{cohort_name}_out",
```
[the remark this answers is not in the document]
Comment (rdmorin) [reply modules/rainstorm/1.0/rainstorm.smk:107]: I think I recall Bruno saying that the config is loaded after so the version information wouldn't get used where we need it.

File: modules/rainstorm/1.0/rainstorm.smk [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+# Generates the target sentinels for each run, which generate the symlinks
+rule _rainstorm_all:
+    input:
+        expand(
+            [
+                str(rules._rainstorm_output_tsv.output.tsv),
+            ],
+            zip,  # Run expand() with zip(), not product()
+            genome_build=CFG["options"]["genome_build"],
```
[the remark this answers is not in the document]
Comment (rdmorin) [reply modules/rainstorm/1.0/rainstorm.smk:209]: The merged maf uses a mix of lifted over and non-lifted so even this information becomes irrelevant per cohort

File: modules/rainstorm/1.0/rainstorm.smk [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+##### RULES #####
+
+# download rainstorm files from git repo without cloning the repo itself
+# decompress files into the 00-inputs
+rule _rainstorm_install:
+    output:
+        rainstorm = CFG["dirs"]["inputs"] + "mutation_rainstorm-0.3/rainstorm.py", # one of the files in the repo
+        peaks = CFG["dirs"]["inputs"] + "mutation_rainstorm-0.3/rainstorm_peaks.R" # another script from the repo
+    params:
+        url = "https://github.com/rdmorin/mutation_rainstorm/archive/refs/tags/v0.3.tar.gz",
```
[the remark this answers is not in the document]
Comment (Kdreval) [reply modules/rainstorm/1.0/rainstorm.smk]: I think it is a good suggestion. I will use 0.3 as default, but make it possible to modify if the future updates will have never tags

File: modules/rainstorm/1.0/rainstorm.smk (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kdreval) [reply modules/rainstorm/1.0/rainstorm.smk:109]: Yes, there are other flags as well. Will add parameter here!

**Comment by Kdreval** [comment]
I have allowed to specify rainstorm version in config, add more flags during the run, removed schema while keeping the oncopipe's setup module working, and also added rule that converts output files to bed. It is ready for re-review.

**Review by lkhilton — APPROVED** [review]

**Pull request merged**